assets/js/base/hooks/shipping/use-shipping-address.js

```diff
@@ -13,22 +13,22 @@ import { CART_STORE_KEY as storeKey } from '@woocommerce/block-data';
 import { useStoreCart } from '../cart/use-store-cart';
 import { pluckAddress } from '../../utils';
 
-const reducer = ( state, address ) =>
-	isShallowEqual( pluckAddress( address ), pluckAddress( state ) )
-		? state
-		: { ...state, ...address };
+const shouldUpdateStore = ( oldAddress, newAddress ) =>
+	! isShallowEqual( pluckAddress( oldAddress ), pluckAddress( newAddress ) );
 
 export const useShippingAddress = () => {
 	const { shippingAddress: initialAddress } = useStoreCart();
-	const [ shippingAddress, setShippingAddress ] = useReducer(
-		reducer,
-		initialAddress
-	);
+	const [ shippingAddress, setShippingAddress ] = useState( initialAddress );
```

Comment:
I usually prefer `useReducer` for form handling, but since we didn’t update the checkout form updating method (still works as if this is `useState`) I think this is fine, we do have one use case that could benefit from `useReducer` (updating state select when you a country is set) but it feels like over engineering, so this is good.

Reply:
`useReducer` is mostly for when you have multiple conditions in which the state changes, or when the state has different slices that get updated instead of the _entire state_.  Since we're dealing with a simple update replacing the entire state, `useReducer` for this use-case was definitely overkill.